server/src/test/java/com/defold/extender/services/DefoldSDKServiceTest.java

```diff
@@ -57,20 +80,86 @@ public void onlyStoreTheNewest() throws IOException, URISyntaxException, Extende
     }
 
     @Test
-    public void testGetSDK() throws IOException, URISyntaxException, ExtenderException {
-        DefoldSdkService defoldSdkService = new DefoldSdkService("/tmp/defoldsdk", 3, true, mock(MeterRegistry.class));
+    public void testGetSDK() throws IOException, ExtenderException {
+        DefoldSdkService defoldSdkService = new DefoldSdkService(folderPath, 3, true, mock(MeterRegistry.class));
 
         File dir = new File("/tmp/defoldsdk/notexist");
         assertFalse(Files.exists(dir.toPath()));
 
-        {
-            boolean thrown = false;
-            try {
-                defoldSdkService.getSdk("notexist");
-            } catch (ExtenderException e) {
-                thrown = true;
-            }
-            assertTrue(thrown);
+        assertThrows(ExtenderException.class, () -> defoldSdkService.getSdk("notexist"));
+    }
+
+    @Test
+    public void testGetSDKRefCount() throws IOException, ExtenderException, InterruptedException {
+        final String testSdk = "11d2cd3a9be17b2fc5a2cb5cea59bbfb4af1ca96";
+        final int expectedRefCount = 3;
+        List<DefoldSdk> sdks = new ArrayList<>();
+        DefoldSdkService defoldSdkService = new DefoldSdkService(folderPath, 0, true, new SimpleMeterRegistry());
+
+        // check when several threads request one sdk and that sdk need to be downloaded
+        ExecutorService service = Executors.newFixedThreadPool(10);
+        CountDownLatch latch = new CountDownLatch(expectedRefCount);
+        for (int i = 0; i < expectedRefCount; ++i) {
```

Comment:
Here simulate case when several threads arrived more or less at the same time and request sdk to download.